path_rotate_root.go

```diff
@@ -0,0 +1,256 @@
+package azuresecrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault-plugin-secrets-azure/api"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/parseutil"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/mitchellh/mapstructure"
+)
+
+func pathRotateRoot(b *azureSecretBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: "rotate-root",
+		Fields: map[string]*framework.FieldSchema{
+			"expiration": {
+				Type: framework.TypeString,
+				// 28 weeks (~6 months) -> days -> hours
+				Default:     (28 * 7 * 24 * time.Hour).String(),
+				Description: "The expiration date of the new credentials in Azure. This can be either a number of seconds or a time formatted duration (ex: 24h)",
+				Required:    false,
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathRotateRoot,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+			},
+		},
+
+		HelpSynopsis: "Attempt to rotate the root credentials used to communicate with Azure.",
+		HelpDescription: "This path will attempt to generate new root credentials for the user used to access and manipulate Azure.\n" +
+			"The new credentials will not be returned from this endpoint, nor the read config endpoint.",
+	}
+}
+
+func (b *azureSecretBackend) pathRotateRoot(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	expirationDur, err := parseutil.ParseDurationSecond(data.Get("expiration").(string))
+	if err != nil {
+		return nil, fmt.Errorf("invalid expiration: %w", err)
+	}
+	expiration := time.Now().Add(expirationDur)
+
+	config, err := b.getConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	passCred, warnings, err := b.rotateRootCredentials(ctx, req.Storage, *config, expiration)
+	if err != nil {
+		return nil, err
+	}
+
+	resultData := map[string]interface{}{
+		"secret_id": *passCred.KeyID,
+		"end_date":  *passCred.EndDate,
+	}
+	if passCred.DisplayName != nil && *passCred.DisplayName != "" {
+		resultData["display_name"] = *passCred.DisplayName
+	}
+
+	resp := &logical.Response{
+		Data:     resultData,
```

Comment:
I don't see anywhere else where we return results in a rotate root. Looks like this on my end:
```
Key             Value
---             -----
display_name    vault-ae443608-693b-6804-b463-f215f09396bc
end_date        2022-04-22T20:28:19.459986Z
secret_id       0ba38415-a58d-4983-b7f1-4cb6da68080f
```

None of this seems particular useful, so I'm wondering if we should just remove the data from the payload. Thoughts?